### PR TITLE
Add validate payload helpers

### PR DESCRIPTION
**Pull request by terencechain**
Review and merge #10024 first

This PR adds validate_payload helpers as defined in the spec

### CODE REVIEW

File: beacon-chain/core/blocks/payload_test.go [hunk 1]
```diff
@@ -409,6 +413,106 @@ func Test_ExecutionEnabled(t *testing.T) {
...
+		payload *ethpb.ExecutionPayload
+		header  *ethpb.ExecutionPayloadHeader
+		err     error
+	}{
+		{
+			name:    "merge incomplete",
+			payload: emptyPayload(),
+			header:  emptyPayloadHeader(),
+			err:     nil,
+		},
```
Comment (rkapka) [inline beacon-chain/core/blocks/payload_test.go:428]: How does this setup ensure that we are pre-merge?

File: beacon-chain/core/blocks/payload_test.go [hunk 2]
```diff
@@ -409,6 +413,106 @@ func Test_ExecutionEnabled(t *testing.T) {
...
+		{
+			name:    "validate passes",
+			payload: emptyPayload(),
+			header: func() *ethpb.ExecutionPayloadHeader {
+				h := emptyPayloadHeader()
+				h.ParentHash = bytesutil.PadTo([]byte{'a'}, fieldparams.RootLength)
```
Comment (rkapka) [inline beacon-chain/core/blocks/payload_test.go:434]: Empty payload and payload header with `a` parent hash are equal?

File: beacon-chain/core/blocks/payload_test.go [hunk 3]
```diff
@@ -409,6 +413,106 @@ func Test_ExecutionEnabled(t *testing.T) {
...
+			name: "incorrect blockhash",
+			payload: func() *ethpb.ExecutionPayload {
+				h := emptyPayload()
+				h.ParentHash = bytesutil.PadTo([]byte{'a'}, fieldparams.RootLength)
+				return h
+			}(),
+			header: func() *ethpb.ExecutionPayloadHeader {
+				h := emptyPayloadHeader()
+				h.ParentHash = bytesutil.PadTo([]byte{'a'}, fieldparams.RootLength)
+				return h
```
Comment (rkapka) [inline beacon-chain/core/blocks/payload_test.go:449]: 1. You should set the payload's `BlockHash`
2. These two should not be equal. Currently both are `a`

File: beacon-chain/core/blocks/payload_test.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/core/blocks/payload_test.go:428]: https://github.com/prysmaticlabs/prysm/pull/10028/files#diff-abcfccaaeba0eafc1bfaf19006a05c7591ec4711e053781094f1e68b7aa9ebaaR28

File: beacon-chain/core/blocks/payload_test.go (hunk 2 above)
Comment (terencechain) [reply beacon-chain/core/blocks/payload_test.go]: Ah. The test setup is bogus. `ParentHash` is not used. It was just comparing zeros so it passed

**Review by rkapka — APPROVED** [review]

**Pull request merged**